exclude mutated buffer

cccclai opened the pull request:
Differential Revision: D55812844

YifanShenSZ commented on exir/backend/utils.py:
```diff
@@ -515,6 +515,12 @@ def tag_constant_data(edge_program: ExportedProgram) -> None:
             or is_buffer(edge_program, node)
             or is_lifted_tensor_constant(edge_program, node)
         ):
+            for node_user in node.users:
```
Shall we make this an option, rather than hard coded?

Concretely, shall we revise line 503 to
```
def tag_constant_data(edge_program: ExportedProgram, exclude_mutable_buffer: bool = False) -> None:
```
so it is still backward compatible, while give us the flexibility to exclude mutable buffer

cccclai replied:
This util function is to tag constant data, while mutable buffer isn't really considered as a constant. I think the backend will know more precisely what kind of mutate operator they can support, and the custom logic can live inside the partitioner.

cccclai committed:
exclude mutated buffer (#2876)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/2876

Differential Revision: D55812844

cccclai commented:
@pytorchbot cherry-pick --onto release/0.2 -c critical
